### PR TITLE
Add various fixes

### LINKED ISSUE #58: Revisit default value of authDatabase
Currently, `authDatabase` defaults to `admin`. It needs to default to whatever database name is being used, or better yet, we should stop adding it to the connection string altogether unless the property is actively set in the configuration.

### PR DESCRIPTION
**Pull request by eduardoboucas**
Closes #58.

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -1,79 +1,112 @@
...
+Object.keys(databases).forEach(databaseName => {
+  const databaseConfig = convict(DATABASE_SCHEMA)
+
+  databaseConfig.load(databases[databaseName])
+  databaseConfig.validate()
+
+  const schema = databaseConfig.getSchema().properties
+
+  // Listening for database-specific environment variables.
+  // e.g. DB_testdb_USERNAME
```
Comment (jimlambie) [inline config.js]: I wonder if we should enforce capitalisation here, so the env var might be `DB_{database.toUpperCase()}_USERNAME`

Comment (eduardoboucas) [reply config.js]: Agree! Will change.

File: lib/index.js [hunk 2]
```diff
@@ -55,18 +58,18 @@ util.inherits(DataStore, EventEmitter)
  * @param {ConnectionOptions} options
  */
 DataStore.prototype.connect = function (options) {
-  this.connectionOptions = this.getConnectionOptions(options)
-  this.connectionString = this.constructConnectionString(this.connectionOptions)
+  const {name, options: databaseBlock} = this.getDatabaseOptions(options)
+  const connectionString = this.getConnectionString(name, databaseBlock)
 
-  debug('connect %s %o', this.connectionString, options)
+  debug('Connect %s %o', connectionString, options)
```
Comment (jimlambie) [inline lib/index.js]: Only if you're sure... the convention is lowercase, as the debug line starts with the whatever is specified when `debug` is required at the top of the file, e.g. `require('debug')('api-mongodb')`

Comment (jimlambie) [reply lib/index.js]: `api-mongodb: connect etc`

Comment (eduardoboucas) [inline lib/index.js]:
```suggestion
  debug('connect %s %o', connectionString, options)
```

**Review by jimlambie — APPROVED** [review]

**Review by jimlambie — APPROVED** [review]
🌮 🌮

**Pull request merged**